### PR TITLE
Fix code search input for different views

### PR DESCRIPTION
**Pull request by wxiaoguang**
Now only show the "code search" on the repo home page, because it only does global search.
So do not show it when viewing file or directory to avoid misleading users (it doesn't search in a directory)

### CODE REVIEW

**Comment by wxiaoguang** [comment]
When debugging, I also removed the unused `IsRepoToolbarCommits` by the way, and add some comments.

<details>

image

image

image

</details>

**Review by lunny — APPROVED** [review]

File: templates/repo/home.tmpl [hunk 1]
```diff
@@ -136,7 +138,7 @@
 					</div>
 					{{template "repo/cite/cite_modal" .}}
 				{{end}}
-				{{if and (not $isHomepage) (not .IsViewFile) (not .IsBlame)}}
+				{{if and (not $isHomepage) (not .IsViewFile) (not .IsBlame)}}{{/* IsViewDirectory (not home) */}}
```
Comment (silverwind) [inline templates/repo/home.tmpl]: Add a variable for IsViewDirectory?

Comment (wxiaoguang) [reply templates/repo/home.tmpl]: Don't think so, as this is a quick bug fix, do not want to introduce any potential regression.

Comment (yp05327) [reply templates/repo/home.tmpl]: So we need to remember adding `(not .IsXXX)` here when adding a new page?

Comment (wxiaoguang) [reply templates/repo/home.tmpl]: I think we need to rewrite the page at that time, instead of introducing more `if` tricks.

Comment (yp05327) [reply templates/repo/home.tmpl]: Agree. So can you add TODO or FIXME here, for avoiding forgetting this.

**Review by yp05327 — APPROVED** [review]

File: templates/repo/home.tmpl (hunk 1 above)
Comment (wxiaoguang) [reply templates/repo/home.tmpl]:
```suggestion
				{{if and (not $isHomepage) (not .IsViewFile) (not .IsBlame)}}{{/* IsViewDirectory (not home), TODO: split the templates, avoid using "if" tricks */}}
```

Comment (silverwind) [reply templates/repo/home.tmpl]: You forgot to commit the suggestion.

Comment (wxiaoguang) [reply templates/repo/home.tmpl]: Really ? I do not think I forgot. Why I have to keep explaining everything .........

image

Comment (silverwind) [reply templates/repo/home.tmpl]: Oh ok, I missed https://github.com/go-gitea/gitea/pull/30678/commits/b3829ad25c75db883e18ac64155cc06cf7ec9e86. GitHub wasn't showing it in the commit list for some reason.

**Pull request merged**